polus-label-to-vector-plugin/src/main.py

```diff
@@ -0,0 +1,74 @@
+from bfio.bfio import BioReader
+import argparse, logging, sys
+import numpy as np
+from pathlib import Path
+import dynamics
+import zarr
+
+if __name__=="__main__":
+    # Initialize the logger
+    logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                        datefmt='%d-%b-%y %H:%M:%S')
+    logger = logging.getLogger("main")
+    logger.setLevel(logging.INFO)
+
+    ''' Argument parsing '''
+    logger.info("Parsing arguments...")
+    parser = argparse.ArgumentParser(prog='main', description='Cellpose parameters')
+
+    # Input arguments
+    parser.add_argument('--inpDir', dest='inpDir', type=str,
+                         help='Input image collection to be processed by this plugin', required=True)
+    # Output arguments
+    parser.add_argument('--outDir', dest='outDir', type=str,
+                        help='Output collection', required=True)
+
+    # Parse the arguments
+    args = parser.parse_args()
+    inpDir = args.inpDir
+    if (Path.is_dir(Path(args.inpDir).joinpath('images'))):
+        # switch to images folder if present
+        fpath = str(Path(args.inpDir).joinpath('images').absolute())
+    logger.info('inpDir = {}'.format(inpDir))
+    outDir = args.outDir
+    logger.info('outDir = {}'.format(outDir))
+
+    # Surround with try/finally for proper error catching
+    try:
+        # Start  logging
+        logger.info('Initializing ...')
+        # Get all file names in inpDir image collection
+        inpDir_files = [f.name for f in Path(inpDir).iterdir() if f.is_file() and "".join(f.suffixes) == '.ome.tif' ]
+        if Path(outDir).joinpath('flow.zarr').exists():
+           raise FileExistsError('Zarr file exists. Delete the existing file')
+        root = zarr.group(store=str(Path(outDir).joinpath('flow.zarr')))
+        for f in inpDir_files:
+            logger.info('Processing image %s ',f)
+        # Loop through files in inpDir image collection and process
+            br = BioReader(str(Path(inpDir).joinpath(f).absolute()))
+            tile_size = min(1080, br.X)
+            # Initializing a array to store the vector field
+            flow_final = np.zeros((br.Y, br.X, br.Z, 3, 1)).astype(np.float32)
+            # Iterating over Z dimension
+            for z in range(br.Z):
+                for x in range(0, br.X, tile_size):
+                    x_max = min([br.X, x + tile_size])
+                    for y in range(0, br.Y, tile_size):
```

Comment:
It may not be a bad idea to change the stride size to something larger (like 4096), since this does not appear to be a super memory intensive operation. Also, what happens when you have a label that is split across a tile? Will the vectors be constructed in such a way that reconstruction would cause them to be identified as different ROIs? If so, should we process this data using overlapping tiles?